core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -234,7 +234,6 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
     }
     for (stageId <- jobData.stageIds) {
       stageIdToActiveJobIds.get(stageId).foreach { jobsUsingStage =>
-        jobsUsingStage.remove(jobEnd.jobId)
```

Comment:
Thanks @vanzin for looking into this. I agree, the change will prevent to remove entries from the map. I will update the PR with the better solution.